docs/modules/integrate/pages/integrate-with-vertx.adoc

```diff
@@ -0,0 +1,131 @@
+- Distributed Map support
+- Distributed Locks
+- Distributed Counters
+
+There are 2 modules to choose from:
+- `io.vertx:vertx-hazelcast` - this module is part of Vert.x and is maintained by the Vert.x team with contributions from Hazelcast developers. This module is licensed under the Apache 2 license.
+
+- `com.hazelcast:vertx-hazelcast-enterprise` - this module is built on top of `vertx-hazelcast` and leverages functionality of {enterprise-product-name} to implement some of the cluster manager functionality (e.g. it uses the CP Subsystem to implement strongly consistent `io.vertx.core.shareddata.Lock` and `io.vertx.core.shareddata.Counter`).
+
+== Using Vert.x Hazelcast Enterprise Cluster Manager
+
+To use the Vert.x Hazelcast Enterprise Cluster Manager, add the following dependency to your Vert.x application:
+
+[source,xml]
+----
+<dependency>
+  <groupId>com.hazelcast</groupId>
+  <artifactId>vertx-hazelcast-enterprise</artifactId>
+  <version>5.0.0-SNAPSHOT</version>
+</dependency>
+----
+
+The dependency is located in the Hazelcast private repository, and you need to add that as well:
+
+[source,xml]
+----
+<repositories>
+  <repository>
+    <id>hazelcast-private-repository</id>
+    <name>Hazelcast Private Repository</name>
+    <url>https://repository.hazelcast.com/release/</url>
+  </repository>
+</repositories>
+----
+
+Alternatively, if you need to use a snapshot version:
+
+[source,xml]
+----
+<repositories>
+  <repository>
+    <id>hazelcast-private-snapshot-repository</id>
+    <name>Hazelcast Private Snapshot Repository</name>
+    <url>https://repository.hazelcast.com/snapshot/</url>
+  </repository>
+</repositories>
+----
+
+To enable clustering, start your Vert.x application with the `-cluster` parameter.
+
+=== Configuration
+
+Provide a file named `cluster.xml` on your classpath to configure a Hazelcast instance used by Vert.x.
+
+To take advantage of the Lock and Counter data structures backed by the CP subsystem, you need to enable the xref:cp-subsystem:cp-subsystem.adoc[CP Subsytem].
+
+For other configuration methods see the link:https://vertx.io/docs/vertx-hazelcast/java/#configcluster[Vert.x documentation].
+
+=== Versioning and Hazelcast compatibility
+
+The Vert.x Hazelcast Enterprise module follows the versioning of Vertx. If you use an `x.y.z` version of Vert.x, you should use an `x.y.z` version of `vertx-hazelcast-enteprise`.
+
+The `vertx-hazelcast-enteprise` module is compatible with all Hazelcast versions supported at the time of the release of the `vertx-hazelcast-enteprise` module unless stated otherwise.
+
+While older versions may work, such configurations are not supported.
+
+== Using Vert.x Hazelcast Cluster Manager
+
+See the Vert.x Hazelcast Cluster Manager site for reference documentation of the `vertx-hazelcast` module.
+
+You can also follow our xref:vertx-step-by-step-guide.adoc[step-by-step guide].
```

Comment:
Suggest changing the name of the guide to "Get started with Vert.x guide"

Reply:
Done f2a27287